devicetypes/ezex/somfy-blind-controller-ezex.src/somfy-blind-controller-ezex.groovy

```diff
@@ -0,0 +1,120 @@
+ */
+ 
+metadata {
+    definition(name: "SOMFY Blind Controller/eZEX", namespace: "eZEX", author: "eZEX Corp", vid:"SmartThings-smartthings-Somfy_WindowShade", ocfDeviceType: "oic.d.blind") {
+        capability "Actuator"
+        capability "Configuration"
+        capability "Refresh"
+        capability "Window Shade"
+        capability "Health Check"
+        capability "Switch Level"
+        
+        command "pause"
+
+        fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0102", outClusters: "0019", deviceJoinName: "SOMFY Blind Controller/eZEX", model: "E2B0-KR000Z0-HA"// SY-IoT201-BD
+    }
+
+
+    tiles(scale: 2) {
+         multiAttributeTile(name:"windowShade", type: "generic", width: 6, height: 4){
+            tileAttribute ("device.windowShade", key: "PRIMARY_CONTROL") {
+              attributeState "open", label:'Open', action:"close", icon:"http://www.ezex.co.kr/img/st/window_open.png", backgroundColor:"#00A0DC", nextState:"turningOff"
+              attributeState "close", label:'Closed', action:"open", icon:"http://www.ezex.co.kr/img/st/window_close.png", backgroundColor:"#ffffff", nextState:"turningOn"
+              attributeState "turningOn", label:'Opening',  action:"close", icon:"http://www.ezex.co.kr/img/st/window_open.png", backgroundColor:"#00A0DC", nextState:"turningOff"
+              attributeState "turningOff", label:'Closing', icon:"http://www.ezex.co.kr/img/st/window_close.png", backgroundColor:"#ffffff", nextState:"turningOn"  
+              //attributeState "partially open", label:'Partially open', action: "switch.off", icon:"http://www.ezex.co.kr/img/st/window_open.png", backgroundColor:"#d45614", nextState: "turningOff"
+            }         
+        }
+        standardTile("home", "device.level", width: 2, height: 2, decoration: "flat") {
+          state "default", label: "home", action:"presetPosition", icon:"st.Home.home2"
+        }
+        standardTile("contPause", "device.switch", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+        	state "pause", label:"", icon:'st.sonos.pause-btn', action:'pause', backgroundColor:"#cccccc"         
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 1) {
+          state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }      
+        valueTile("ShadeLevel", "device.level", width: 4, height: 1) {
+        	state "level", label: 'Shade is ${currentValue}% up', defaultState: true
+        }
+        
+        controlTile("levelSliderControl", "device.level", "slider", width:2, height: 1, inactiveLabel: false) {
+        	state "level", action:"switch level.setLevel"
+        }
+        
+        preferences {
+          input "preset", "number", title: "Default half-open position (1-100). Springs Window Fashions users should consult their manuals.", defaultValue: 50, required: false, displayDuringSetup: false
```

Comment:
This preference is poorly copied from the Springs DTH. Additionally, there is no update method to handle this value changing. Nor is there a method that uses this value. If the device actually supports a preset position, it should be determined whether it is set via hardware manipulation of the device or if there is some zigbee command that needs to be sent.

Reply:
As I commented in about the preset above, I will add after it is clear. This device cannot be set to go to the presetPosition via H/W button but available via zigbee command like below.
 zigbee.command(0x0102, 0x05, zigbee.convertToHexString(data, 2))

Reply:
@greens , as @tpmanley answered above, this device doesn't seem to support presetPosition and I will remove this preference() too.